### DISCUSSION TITLE
Can't get it working on UDM pro with XS4ALL / KPN

### DISCUSSION THREAD

**Discussion by MeeevA** [fabianishere/udm-iptv#249] [Q&A]
Hello, I had a working situation with an old firmware version of the UDM. Now after upgrading can not get it to work anymore.

What I changed was also that I now have a direct fiber connection into the UDM Pro and not with the fiber to UTP NTU anymore.

Internet works fine, but IPTV not. Message I got was that the settop box can not reach the videoserver. That was when it was still not rebooted.

I tried several things amongst reconfiguring (from UDP port to Fiber), uninstall and install again, but unfortunately nothing.

My diagnose is:
=== Service Logs ===
Apr 16 14:01:35 UDMProGRAS udm-iptvd[3041]: FATAL[init_interface@251]: exiting.....
Apr 16 14:01:35 UDMProGRAS udm-iptvd[3041]: ERROR[main@619]: exiting.........
Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Main process exited, code=exited, status=1/FAILURE
Apr 16 14:01:35 UDMProGRAS IMPROXY[3041]: ERROR[init_interface@233]: Disable IGMP, there is not igmp upstream interface or igmp downstream interface
Apr 16 14:01:35 UDMProGRAS IMPROXY[3041]: FATAL[init_interface@251]: exiting.....
Apr 16 14:01:35 UDMProGRAS IMPROXY[3041]: ERROR[main@619]: exiting.........
Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Consumed 145ms CPU time.
Apr 16 14:01:35 UDMProGRAS systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Consumed 145ms CPU time.
================

Can somebody point me in the right direction.

**Comment by fabianishere** [comment]
You are missing part of the diagnosis. I suspect you are using the wrong WAN port based on the error log.

**Comment by MeeevA** [comment]
Hello Fabian,

Thank you for your quick reply.

[image: image.png]

Just to be 100% sure I did another reconfigure and for sure I select the
right port (wan 2 Fiber)

But I did note now when finalising the following:

Failed to preset unit: Unit file /etc/systemd/system/udm-iptv.service is
masked.

/usr/bin/deb-systemd-helper: error: systemctl preset failed on
udm-iptv.service: No such file or directory

udm-iptv.service is a disabled or a static unit not running, not starting
it.

I think this might explain. Can you help me what is wrong?

***@***.***:~# udm-iptv diagnose

Please share the following output with the developers:

=== Configuration ===

WAN Interface: eth9

WAN VLAN: 4 (dev iptv)

WAN DHCP: true (options "-O staticroutes -V IPTV_RG")

WAN Ranges: 213.75.0.0/16  217.166.0.0/16

LAN Interfaces:

IGMP Proxy quickleave disabled: false

IGMP Proxy debug: true

=== IP Link and Route ===

Device "iptv" does not exist.

Cannot find device "iptv"

=== Service Logs ===

Apr 16 14:01:35 UDMProGRAS udm-iptvd[3041]: ***@***.***:
exiting.....

Apr 16 14:01:35 UDMProGRAS udm-iptvd[3041]: ***@***.***:
exiting.........

Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Main process
exited, code=exited, status=1/FAILURE

Apr 16 14:01:35 UDMProGRAS IMPROXY[3041]: ***@***.***:
Disable IGMP, there is not igmp upstream interface or igmp downstream
interface

Apr 16 14:01:35 UDMProGRAS IMPROXY[3041]: ***@***.***:
exiting.....

Apr 16 14:01:35 UDMProGRAS IMPROXY[3041]: ***@***.***: exiting.........

Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Failed with result
'exit-code'.

Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Consumed 145ms CPU
time.

Apr 16 14:01:35 UDMProGRAS systemd[1]: Stopped IPTV support for the UniFi
Dream Machine.

Apr 16 14:01:35 UDMProGRAS systemd[1]: udm-iptv.service: Consumed 145ms CPU
time.

***@***.***:~#


Op ma 17 apr 2023 om 09:39 schreef Fabian Mastenbroek <
***@***.***>:

**Reply by MeeevA** [reply]
After some more looking I found that udm-iptv.service was masked. I did a complete uninstall and deleted the remaining files. After that I rebooted and installed again.

Now it is still not working, but getting closer I think.

Can you please point me in the right direction.

Below the current diagnose:

Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth9
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: 
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
55: iptv@eth9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.180.31.162/22 brd 10.180.31.255 scope global iptv
       valid_lft forever preferred_lft forever
10.180.28.0/22 proto kernel scope link src 10.180.31.162 
213.75.112.0/21 via 10.180.28.1 metric 255 
=== Service Logs ===
Apr 17 16:30:10 UDMProGRAS udm-iptvd[104747]: RECV V2 member report   from 192.168.1.173   to 224.0.0.251
Apr 17 16:30:10 UDMProGRAS udm-iptvd[104747]: Mebership report was received on a disabled interface. Ignoring.
Apr 17 16:30:34 UDMProGRAS udm-iptvd[104747]: RECV V2 member report   from 192.168.1.223   to 224.0.0.251
Apr 17 16:30:34 UDMProGRAS udm-iptvd[104747]: Mebership report was received on a disabled interface. Ignoring.
Apr 17 16:31:04 UDMProGRAS udm-iptvd[104747]: RECV V2 member report   from 192.168.1.223   to 224.0.0.251
Apr 17 16:31:04 UDMProGRAS udm-iptvd[104747]: Mebership report was received on a disabled interface. Ignoring.
Apr 17 16:31:07 UDMProGRAS udm-iptvd[104747]: RECV V2 member report   from 192.168.1.223   to 224.0.0.251
Apr 17 16:31:07 UDMProGRAS udm-iptvd[104747]: Mebership report was received on a disabled interface. Ignoring.
Apr 17 16:31:29 UDMProGRAS udm-iptvd[104747]: RECV V2 member report   from 192.168.1.223   to 224.0.0.251
Apr 17 16:31:29 UDMProGRAS udm-iptvd[104747]: Mebership report was received on a disabled interface. Ignoring.
root@UDMProGRAS:/etc#

**Reply by MeeevA** [reply]
Ohh, lan missing I see. Added and now it is working
